Frameworks/Starboard/EbrFile.cpp

```diff
@@ -161,20 +171,47 @@ bool EbrUnlink(const char* path) {
 }
 
 #define mkdir _mkdir
-std::wstring g_WritableFolder(L".");
 
-void IwSetWritableFolder(const wchar_t* folder) {
-    g_WritableFolder = folder;
-    // recreate the default folders
-    CPathMapper::CreateDefaultPaths();
+Wrappers::HString _IwGetWritableFolder() {
+    Wrappers::HString toReturn;
+    ComPtr<IStorageFolder> folder;
+    ComPtr<IApplicationDataStatics> applicationDataStatics;
+    ComPtr<IApplicationData> applicationData;
+    ComPtr<IStorageItem> storageItem;
+
+    if (FAILED(GetActivationFactory(Wrappers::HStringReference(RuntimeClass_Windows_Storage_ApplicationData).Get(),
+                                    &applicationDataStatics))) {
+        return toReturn;
+    }
+
+    if (FAILED(applicationDataStatics->get_Current(&applicationData))) {
+        return toReturn;
+    }
+
+    if (FAILED(applicationData->get_LocalFolder(&folder))) {
+        return toReturn;
+    }
+
+    if (FAILED(folder.As<IStorageItem>(&storageItem))) {
+        return toReturn;
+    }
+
+    if (FAILED(storageItem->get_Path(toReturn.GetAddressOf()))) {
+        return toReturn;
+    }
+
+    return toReturn;
 }
 
 const wchar_t* IwGetWritableFolder() {
-    return g_WritableFolder.c_str();
-}
+    static Wrappers::HString basePath = _IwGetWritableFolder();
+
+    if (!basePath.IsValid()) {
+        return L".";
+    }
 
-const std::wstring& _IwGetWritableFolder() {
-    return g_WritableFolder;
+    unsigned int rawLength;
+    return WindowsGetStringRawBuffer(basePath.Get(), &rawLength);
```

Comment:
`length` is marked `_Out_opt_` in the header; don't make it populate something you're discarding immediately.